osu.Game/Screens/Utility/LatencyCertifierScreen.cs

```diff
@@ -0,0 +1,436 @@
+                        Font = OsuFont.Torus.With(size: 24, weight: FontWeight.SemiBold),
+                        Y = 80,
+                    }
+                });
+
+                background.FadeInFromZero(1000, Easing.OutQuint);
+
+                certifiedText.FadeInFromZero(500, Easing.InQuint);
+
+                certifiedText
+                    .ScaleTo(10)
+                    .ScaleTo(1, 600, Easing.InQuad)
+                    .Then()
+                    .ScaleTo(1.05f, 10000, Easing.OutQuint);
+                return;
+            }
+
+            string cannotIncreaseReason = string.Empty;
+
+            if (!isPass)
+                cannotIncreaseReason = "You didn't get a perfect score.";
+            else if (mapDifficultyToTargetFrameRate(difficultyLevel + 1) > target_host_update_frames)
+                cannotIncreaseReason = "You've reached the limits of this comparison mode.";
+            else if (mapDifficultyToTargetFrameRate(difficultyLevel + 1) > Clock.FramesPerSecond)
+                cannotIncreaseReason = "Game is not running fast enough to test this level";
+
+            resultsArea.Add(new FillFlowContainer
+            {
+                RelativeSizeAxes = Axes.X,
+                AutoSizeAxes = Axes.Y,
+                Anchor = Anchor.BottomLeft,
+                Origin = Anchor.BottomLeft,
+                Spacing = new Vector2(20),
+                Padding = new MarginPadding(20),
+                Children = new Drawable[]
+                {
+                    new ButtonWithKeyBind(Key.Enter)
+                    {
+                        Text = "Continue to next level",
+                        BackgroundColour = colours.Red2,
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        Action = () => changeDifficulty(difficultyLevel + 1),
+                        Enabled = { Value = string.IsNullOrEmpty(cannotIncreaseReason) },
+                        TooltipText = cannotIncreaseReason
+                    },
+                    new ButtonWithKeyBind(Key.D)
+                    {
+                        Text = difficultyLevel == 1 ? "Retry" : "Return to last level",
+                        BackgroundColour = colours.Green,
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        Action = () => changeDifficulty(Math.Max(difficultyLevel - 1, 1)),
+                    },
+                    new ButtonWithKeyBind(Key.C)
+                    {
+                        Text = $"Continue towards certification at this level ({certificationRemaining} more)",
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        Action = () =>
+                        {
+                            resultsArea.Clear();
+                            targetRoundCount += rounds_to_complete;
```

Comment:
Can `targetRoundCount` be renamed? This threw me off and I wasn't sure if this line wasn't adding even more attempts to reach certification if the level is repeated, while this is doing the exact opposite (as far as I can tell `targetRoundCount` stores the number of choices the user has made up to this point).

Reply:
Actually it's doing what you first thought. Or rather, specifying the number of rounds required to next reach the results screen.

I'll see if I can make it read better.